samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1682,6 +1682,387 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
+		connected = true;
+	}
+
+	link_api_modem_info_get_for_shell(connected);
+	return ret;
+}
+
+enum link_shell_common_options link_shell_getopt_common(int argc, char *const argv[])
+{
+	enum link_shell_common_options common_option = LINK_COMMON_NONE;
+
+	/* Reset getopt due to possible previous failures: */
+	optreset = 1;
+
+	/* We start from subcmd arguments */
+	optind = 1;
+
+	int long_index = 0;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv,
+				  "sured",
+				  long_options, &long_index)) != -1) {
+		switch (opt) {
+
+		case 's':
+			common_option = LINK_COMMON_SUBSCRIBE;
+			break;
+		case 'u':
+			common_option = LINK_COMMON_UNSUBSCRIBE;
+			break;
+		case 'e':
+			common_option = LINK_COMMON_ENABLE;
+			break;
+		case 'd':
+			common_option = LINK_COMMON_DISABLE;
+			break;
+		case 'r':
+			common_option = LINK_COMMON_READ;
+			break;
+		case LINK_SHELL_OPT_RESET:
+			common_option = LINK_COMMON_RESET;
+			break;
+		case LINK_SHELL_OPT_START:
+			common_option = LINK_COMMON_START;
+			break;
+		case LINK_SHELL_OPT_STOP:
+			common_option = LINK_COMMON_STOP;
+			break;
+		case '?':
+			/* TODO: We could print usage here */
+			break;
+		default:
+			break;
+		}
+	}
+	return common_option;
+}
+
+int link_funmode(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret = 0;
+	bool nmode_use_rel14 = true;
+	enum lte_lc_func_mode funmode_option = LINK_FUNMODE_NONE;
+	enum lte_lc_func_mode functional_mode;
+	enum link_shell_common_options common_option;
+	char snum[64];
+
+	common_option = link_shell_getopt_common(argc, argv);
+
+	/* Reset getopt due to possible previous failures */
+	optreset = 1;
+	/* We start from subcmd arguments */
+	optind = 1;
+
+	int long_index = 0;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv,
+				  "014",
+				  long_options, &long_index)) != -1) {
+		switch (opt) {
+		case '0':
+			funmode_option = LTE_LC_FUNC_MODE_POWER_OFF;
+			break;
+		case '1':
+			funmode_option = LTE_LC_FUNC_MODE_NORMAL;
+			break;
+		case '4':
+			funmode_option = LTE_LC_FUNC_MODE_OFFLINE;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_LTEOFF:
+			funmode_option = LTE_LC_FUNC_MODE_DEACTIVATE_LTE;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_LTEON:
+			funmode_option = LTE_LC_FUNC_MODE_ACTIVATE_LTE;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_GNSSOFF:
+			funmode_option = LTE_LC_FUNC_MODE_DEACTIVATE_GNSS;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_GNSSON:
+			funmode_option = LTE_LC_FUNC_MODE_ACTIVATE_GNSS;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_UICCOFF:
+			funmode_option = LTE_LC_FUNC_MODE_DEACTIVATE_UICC;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_UICCON:
+			funmode_option = LTE_LC_FUNC_MODE_ACTIVATE_UICC;
+			break;
+		case LINK_SHELL_OPT_FUNMODE_FLIGHTMODE_UICCON:
+			funmode_option = LTE_LC_FUNC_MODE_OFFLINE_UICC_ON;
+			break;
+		case LINK_SHELL_OPT_NMODE_NO_REL14:
+			funmode_option = LTE_LC_FUNC_MODE_NORMAL;
+			nmode_use_rel14 = false;
+			break;
+		default:
+			break;
```

Comment:
We cannot log error nor print usage here because common options take us here.

Reply:
Use of LINK_COMMON_NONE?